app/models/vm_or_template.rb

```diff
@@ -393,7 +393,7 @@ def self.task_arguments(options)
     when "remove_snapshot", "revert_to_snapshot" then
       [options[:snap_selected]]
     when "create_snapshot" then
-      [options[:name], options[:description], options[:memory]]
+      [options[:name], options[:description], options.fetch(:memory, false)]
```

Comment:
My only problem with this (which I guess is my problem with `.fetch` in general), is that it only handles keys that don't exist.

```ruby
{}.fetch(:a, false)
# => false

{:a => nil}.fetch(:a, false)
# => nil
```

To be safe, this might be better written as `!!options[:memory]` if all you want is a literal boolean.